e2e/test/scenarios/question/notebook-native-preview-sidebar.cy.spec.ts

```diff
@@ -358,3 +396,37 @@ function convertToSql() {
   cy.wait("@dataset");
   cy.findByTestId("native-query-editor").should("be.visible");
 }
+
+type ResizeSidebarCallback = (
+  initialSidebarWidth: number,
+  sidebarWidth: number,
+) => void;
+
+function resizeSidebar(amount: number, cb: ResizeSidebarCallback) {
+  cy.intercept("PUT", "/api/setting/notebook-native-preview-sidebar-width").as(
+    "updateSidebarWidth",
+  );
+  cy.intercept("GET", "/api/session/properties").as("sessionProperties");
+
+  cy.findByTestId("native-query-preview-sidebar").then($sidebar => {
+    const initialSidebarWidth = $sidebar[0].getBoundingClientRect().width;
+
+    const options = {
+      pointer: "mouse" as const,
+      position: "center" as const,
+      button: "left" as const,
+    };
```

Comment:
I know we avoid casting when possible, but realCypressEvents has different type definitions for mousedown, mousemove and mouseup. Casting seemed like an easier (or even more elegant) solution in this case.